src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/illegalidentifiername/Example1.java

```diff
@@ -0,0 +1,34 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="IllegalIdentifierName"/>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.illegalidentifiername;
+
+// xdoc section -- start
+public class Example1 {
+  public static void main(String... args) {
+    var var = 4; // violation
+    int record = 15; // violation
+    String yield = "yield"; // violation
+
+    record Record (Record record){} // 2 violations
+
+    String yieldString = "yieldString"; // ok, part of another word
```

Comment:
`// ok, 'yield' is part of name`

Please apply same change to other cases like this

Reply:
done

Reply:
Your new wording is not clear to me. Let's try
`// ok, 'yield' is part of name not whole name `

Reply:
> Your new wording is not clear to me. Let's try `// ok, 'yield' is part of name not whole name `

Here are few other possibilities :  
`// ok, word 'yield' is not used as an identifier by itself`
or 
`// ok, word 'yield' is used only as part of identifier but not whole`

Reply:
`// ok, word 'yield' is not used as an identifier by itself`

lets use this